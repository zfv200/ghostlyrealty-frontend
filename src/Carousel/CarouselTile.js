import React from 'react'
import styles from './Carousel.css.js'
import { Link } from 'react-router-dom'

const CarouselTile = (props) =>{
  if(props){
    return (
      <Link to={`/houses/${props.id}`} style={{ textDecoration: 'none' }}>
        <li className="di fl pa0 w-100" style={styles.carouselChild}>
          {
            props['image_url'] !== "" ?
            <img className="w-100" style={styles.img} src={props['image_url']} alt={props.name}/>
            :
            <img className="w-100" style={styles.img} src={props.images[0]} alt={props.name}/>
          }
          <div className="caption relative bottom-2 bg-black white h2">
            <p className="relative pa1 ml5">{props.name}</p>
          </div>
        </li>
      </Link>
    )
  }
}

export default CarouselTile
