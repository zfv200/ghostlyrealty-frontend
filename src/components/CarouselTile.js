import React from 'react'

const CarouselTile = (props) =>{
  if(props){
    return (
      <div>
        <img src={props['image-url']} alt={props.name}/>
        <h1>{props.name}</h1>
      </div>
    )
  }
}

export default CarouselTile
