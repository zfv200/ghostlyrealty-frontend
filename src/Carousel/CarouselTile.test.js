import React from 'react'
import ReactDOM from 'react-dom'
import CarouselTile from './CarouselTile'

import { shallow } from 'enzyme'

const house = {
  name: "Winchester Mystery House",
  image_url: "https://media.cntraveler.com/photos/580e6787e62940c66bd2be0d/master/w_820,c_limit/winchester-mystery-house-san-jose-GettyImages-110714519.jpg",
}

it('renders without crashing', ()=>{
  const component = shallow(<CarouselTile {...house}/>)

  expect(component).toMatchSnapshot()
})
