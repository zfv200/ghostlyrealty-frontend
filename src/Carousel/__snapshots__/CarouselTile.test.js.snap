// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CarouselTile
    image_url="https://media.cntraveler.com/photos/580e6787e62940c66bd2be0d/master/w_820,c_limit/winchester-mystery-house-san-jose-GettyImages-110714519.jpg"
    name="Winchester Mystery House"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Image
        alt="Winchester Mystery House"
        as="img"
        className=""
        style={
          Object {
            "height": "100%",
            "width": "1000px",
          }
        }
        ui={true}
      />,
      "replace": false,
      "style": Object {
        "textDecoration": "none",
      },
      "to": "/houses/undefined",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "alt": "Winchester Mystery House",
        "as": "img",
        "className": "",
        "onClick": undefined,
        "src": undefined,
        "style": Object {
          "height": "100%",
          "width": "1000px",
        },
        "ui": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Image
          alt="Winchester Mystery House"
          as="img"
          className=""
          style={
            Object {
              "height": "100%",
              "width": "1000px",
            }
          }
          ui={true}
        />,
        "replace": false,
        "style": Object {
          "textDecoration": "none",
        },
        "to": "/houses/undefined",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "alt": "Winchester Mystery House",
          "as": "img",
          "className": "",
          "onClick": undefined,
          "src": undefined,
          "style": Object {
            "height": "100%",
            "width": "1000px",
          },
          "ui": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
